baybe/surrogates/base.py

```diff
@@ -124,60 +134,80 @@ def _make_target_scaler() -> OutcomeTransform | None:
 
     def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
         """Make the input scaler for transforming computational dataframes."""
-        from sklearn.compose import make_column_transformer
-
-        # Create the composite scaler from the parameter-wise scaler objects
-        transformers = [
-            (
-                "passthrough" if (s := self._make_parameter_scaler(p)) is None else s,
-                [c for c in p.comp_rep_columns if c in searchspace.comp_rep_columns],
-            )
-            for p in searchspace.parameters
-        ]
-        scaler = make_column_transformer(*transformers, verbose_feature_names_out=False)
-
-        scaler.fit(searchspace.comp_rep_bounds)
+        # Create a composite scaler from parameter-wise scaler objects
+        mapping: dict[tuple[int, ...], InputTransform] = {}
+        for p in searchspace.parameters:
+            idxs = searchspace.get_comp_rep_parameter_indices(p.name)
+            factory = self._make_parameter_scaler_factory(p)
+            if factory is None:
+                continue
+            transformer = factory(len(idxs))
+            mapping[idxs] = transformer
+        scaler = ColumnTransformer(mapping)
+
+        # Fit the scaler to the parameter bounds
+        scaler.fit(to_tensor(searchspace.comp_rep_bounds))
 
         return scaler
 
     def _make_output_scaler(
```

Comment:
From the descriptions and the docstrings, I do not fully understand the difference between `target` and `output` resp. the corresponding scalers. I see that they do different things in the code, but the naming is not clear to me.

Reply:
Happy to open discussion here, I was also not clear what is the best naming. Current rationale:
* I used the terminology `parameter`/`target` for `_make_target_scaler` and `_make_parameter_scaler_factory` because there is a direct relation to these objects: the methods get these objects as input and have to decide which scaling to apply to the given object
* However, the overall "input scaler" finally operates on a computational dataframe, which i) no longer contains a single parameter but all of them ii) but not necessary all columns of each parameter. So it is a "processed" object, but the role of this object is that it is – generally speaking – the thing we provide as input to our surrogate before applying scaling internally. That's why I took the mathematical perspective here and simply called it input/output of the surrogate.

Do you have a better idea, perhaps?

Reply:
to me the naming makes sense, input/output = stuff in and out of the model, parameters play into input, targets play into output